From: Andrew Morton <akpm@linux-foundation.org>
Date: Fri, 6 Dec 2024 20:24:17 -0800
Subject: [PATCH] mm-introduce-mmap_lock_speculate_try_beginretry-fix

use read_seqcount_retry() in mmap_lock_speculate_retry(), per Wei Yang

Cc: Liam R. Howlett <Liam.Howlett@Oracle.com>
Cc: Peter Zijlstra <peterz@infradead.org>
Cc: Suren Baghdasaryan <surenb@google.com>
Cc: Wei Yang <richard.weiyang@gmail.com>
Cc: Christian Brauner <brauner@kernel.org>
Cc: David Hildenbrand <david@redhat.com>
Cc: David Howells <dhowells@redhat.com>
Cc: Davidlohr Bueso <dave@stgolabs.net>
Cc: Hillf Danton <hdanton@sina.com>
Cc: Hugh Dickins <hughd@google.com>
Cc: Jann Horn <jannh@google.com>
Cc: Johannes Weiner <hannes@cmpxchg.org>
Cc: Jonathan Corbet <corbet@lwn.net>
Cc: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Cc: Mateusz Guzik <mjguzik@gmail.com>
Cc: Matthew Wilcox <willy@infradead.org>
Cc: Mel Gorman <mgorman@techsingularity.net>
Cc: Michal Hocko <mhocko@suse.com>
Cc: Minchan Kim <minchan@google.com>
Cc: Oleg Nesterov <oleg@redhat.com>
Cc: Pasha Tatashin <pasha.tatashin@soleen.com>
Cc: Paul E. McKenney <paulmck@kernel.org>
Cc: Peter Xu <peterx@redhat.com>
Cc: Shakeel Butt <shakeel.butt@linux.dev>
Cc: Sourav Panda <souravpanda@google.com>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 include/linux/mmap_lock.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/mmap_lock.h b/include/linux/mmap_lock.h
index 8ac3041df053..45a21faa3ff6 100644
--- a/include/linux/mmap_lock.h
+++ b/include/linux/mmap_lock.h
@@ -101,7 +101,7 @@ static inline bool mmap_lock_speculate_try_begin(struct mm_struct *mm, unsigned
 
 static inline bool mmap_lock_speculate_retry(struct mm_struct *mm, unsigned int seq)
 {
-	return do_read_seqcount_retry(&mm->mm_lock_seq, seq);
+	return read_seqcount_retry(&mm->mm_lock_seq, seq);
 }
 
 #else /* CONFIG_PER_VMA_LOCK */
-- 
2.39.5

